[ResourceMonitor] Update unloaded frame message.

basuke opened the pull request:
#### bc6a2f20c264d530c26d0fe58b3418553de97f02
<pre>
[ResourceMonitor] Update unloaded frame message.
<a href="https://bugs.webkit.org/show_bug.cgi?id=284857">https://bugs.webkit.org/show_bug.cgi?id=284857</a>
<a href="https://rdar.apple.com/141648202">rdar://141648202</a>

Reviewed by Ben Nham.

Replace current placeholder message with more descriptive and responsive design
for various sizes. Also add custom message for Cocoa platform.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::networkUsageDidExceedThreshold):
(WebCore::generateResourceMonitorErrorHTML):

Canonical link: <a href="https://commits.webkit.org/288044@main">https://commits.webkit.org/288044@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1432973504aa7162564954009ae0f4ec484dd142

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81737 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1263 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35691 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86282 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32733 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/83843 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1296 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9083 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63768 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21490 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84807 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/939 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74383 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44054 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/839 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28558 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31186 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87720 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8977 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/6377 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72104 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9162 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70203 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71333 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17770 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15420 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14339 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8928 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/14460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8769 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12292 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10577 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bnham commented on Source/WebCore/page/LocalFrame.cpp:
```diff
@@ -1393,12 +1397,43 @@ void LocalFrame::networkUsageDidExceedThreshold()
         document->addConsoleMessage(MessageSource::ContentBlocker, MessageLevel::Error, "Frame was unloaded because its network usage exceeded the limit."_s);
 }
 
+String LocalFrame::generateResourceMonitorErrorHTML() const
```
nit: seems like this could just be a `static` function in this file and doesn't need to touch `LocalFrame.h`.

bnham commented on Source/WebCore/page/LocalFrame.cpp:
```diff
@@ -1393,12 +1397,43 @@ void LocalFrame::networkUsageDidExceedThreshold()
+        "<span class=L>⛔︎ Frame hidden for using significant resources</span>"
+        "<span class=M>⛔︎<br />Frame hidden</span>"
+        "<span class=S>⛔︎</span>"
+        "</a>");
```
I think we can leave out the fancy HTML for non-Cocoa platforms; if they implement this feature, they can figure out the design that works best for them and change this function as needed.

Another convention is that any string displayed in the UI should go through the WEB_UI_STRING macro, so that our localization scripts (and the localization scripts for ports) know that this is a string that has to be translated to other languages.

So I think I'd just replace this block with something like `return WEB_UI_STRING("This frame is hidden for using too many system resources.", "Heavy frame unloading description")`. You may also need to include `LocalizedStrings.h`.

bnham replied:
Also, add the localized string key (the first argument to `WEB_UI_STRING`) to `Source/WebCore/en.lproj/Localizable.strings`.

bnham reviewed (approved):
Approved with some comments.

basuke committed:
[ResourceMonitor] Update unloaded frame message.
https://bugs.webkit.org/show_bug.cgi?id=284857
rdar://141648202

Reviewed by Ben Nham.

Replace current placeholder message with more descriptive and responsive design
for various sizes. Also add custom message for Cocoa platform.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::networkUsageDidExceedThreshold):
(WebCore::generateResourceMonitorErrorHTML):

Canonical link: https://commits.webkit.org/288044@main
